### ISSUE TITLE
Unable to generate reports (ERROR 500: Call to a member function date() on null)

### ISSUE THREAD

**Issue by pryley** [statamic/seo-pro#116]
This error also caused the /cp/dashboard to crash with a 500 error when the SEO widget was being used on the dashboard.

Update: **I fixed it** by manually deleting all existing reports, but unsure if this was related to (maybe a side effect of) #111 (see: https://github.com/statamic/seo-pro/commit/abd1697cfbc38c09f285211a424ae2a437a065f3#commitcomment-41742651)

![image](https://user-images.githubusercontent.com/134939/91210439-d6d49c80-e6d2-11ea-893a-4274bde39a3c.png)

```
[2020-08-25 17:55:42] production.ERROR: Call to a member function date() on null {"exception":"[object] (Error(code: 0): Call to a member function date() on null at /home/production/domain.com/releases/1/vendor/laravel/framework/src/Illuminate/Support/HigherOrderCollectionProxy.php:60)
[stacktrace]
#0 /home/production/domain.com/releases/1/vendor/laravel/framework/src/Illuminate/Support/Collection.php(1123): Illuminate\\Support\\HigherOrderCollectionProxy->Illuminate\\Support\\{closure}()
#1 /home/production/domain.com/releases/1/vendor/laravel/framework/src/Illuminate/Support/Collection.php(1148): Illuminate\\Support\\Collection->sortBy()
#2 /home/production/domain.com/releases/1/vendor/laravel/framework/src/Illuminate/Support/HigherOrderCollectionProxy.php(61): Illuminate\\Support\\Collection->sortByDesc()
#3 /home/production/domain.com/releases/1/vendor/statamic/seo-pro/src/Reporting/Report.php(238): Illuminate\\Support\\HigherOrderCollectionProxy->__call()
#4 /home/production/domain.com/releases/1/vendor/statamic/seo-pro/src/Reporting/Report.php(244): Statamic\\SeoPro\\Reporting\\Report::all()
#5 /home/production/domain.com/releases/1/vendor/statamic/seo-pro/src/Reporting/Report.php(55): Statamic\\SeoPro\\Reporting\\Report::latest()
#6 /home/production/domain.com/releases/1/vendor/statamic/seo-pro/src/Reporting/Report.php(41): Statamic\\SeoPro\\Reporting\\Report::nextId()
#7 /home/production/domain.com/releases/1/vendor/statamic/seo-pro/src/Commands/GenerateReportCommand.php(38): Statamic\\SeoPro\\Reporting\\Report::create()
#8 [internal function]: Statamic\\SeoPro\\Commands\\GenerateReportCommand->handle()
#9 /home/production/domain.com/releases/1/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(37): call_user_func_array()
#10 /home/production/domain.com/releases/1/vendor/laravel/framework/src/Illuminate/Container/Util.php(37): Illuminate\\Container\\BoundMethod::Illuminate\\Container\\{closure}()
#11 /home/production/domain.com/releases/1/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(95): Illuminate\\Container\\Util::unwrapIfClosure()
#12 /home/production/domain.com/releases/1/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(39): Illuminate\\Container\\BoundMethod::callBoundMethod()
#13 /home/production/domain.com/releases/1/vendor/laravel/framework/src/Illuminate/Container/Container.php(596): Illuminate\\Container\\BoundMethod::call()
#14 /home/production/domain.com/releases/1/vendor/laravel/framework/src/Illuminate/Console/Command.php(134): Illuminate\\Container\\Container->call()
#15 /home/production/domain.com/releases/1/vendor/symfony/console/Command/Command.php(258): Illuminate\\Console\\Command->execute()
#16 /home/production/domain.com/releases/1/vendor/laravel/framework/src/Illuminate/Console/Command.php(121): Symfony\\Component\\Console\\Command\\Command->run()
#17 /home/production/domain.com/releases/1/vendor/symfony/console/Application.php(911): Illuminate\\Console\\Command->run()
#18 /home/production/domain.com/releases/1/vendor/symfony/console/Application.php(264): Symfony\\Component\\Console\\Application->doRunCommand()
#19 /home/production/domain.com/releases/1/vendor/symfony/console/Application.php(140): Symfony\\Component\\Console\\Application->doRun()
#20 /home/production/domain.com/releases/1/vendor/laravel/framework/src/Illuminate/Console/Application.php(93): Symfony\\Component\\Console\\Application->run()
#21 /home/production/domain.com/releases/1/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(129): Illuminate\\Console\\Application->run()
#22 /home/production/domain.com/releases/1/please(37): Illuminate\\Foundation\\Console\\Kernel->handle()
#23 {main}
"}
```

Statamic 3.0.0 Pro
Laravel 7.26.0
PHP 7.4.9
aerni/social-links 2.0.1
aerni/statamic-imagekit 2.0.1
aryehraber/statamic-impersonator 2.0.0
aryehraber/statamic-splash 1.0.3
doublethreedigital/duplicator 1.0.5
rias/statamic-color-swatches 2.0.5
spatie/statamic-responsive-images 1.2.3
statamic/seo-pro 2.0.8

**Comment by jesseleite** [comment]
Are you able to predictably reproduce this, or is it just happening with random reports?  And when it happens, what does your `storage/statamic/seopro/reports` folder look like? ie)

![image](https://user-images.githubusercontent.com/5187394/91222073-aa2d7e80-e6ec-11ea-88ab-c5a6ca7de1e3.png)

**Comment by jesseleite** [comment]
I think maybe the issue is with how you are manually deleting the report(s).  If you leave a report's folder contents, but leave the folder itself, it'll throw that error.  That said, just pushed a fix to filter out bad reports.  Thanks!

**Comment by pryley** [comment]
Since I have a mirror of the site, I checked to see if I could reproduce:

**App Details:**

Statamic 3.0.0 Pro
Laravel 7.26.0
PHP 7.4.9
aerni/social-links 2.0.1
aerni/statamic-imagekit 2.0.1
aryehraber/statamic-impersonator 2.0.0
aryehraber/statamic-splash 1.0.3
doublethreedigital/duplicator 1.0.5
rias/statamic-color-swatches 2.0.5
spatie/statamic-responsive-images 1.2.3
statamic/seo-pro 2.0.7

### 1. The site does not yet have any reports generated

I also checked the reports dir which was empty, and the SEO Pr addon is at version 2.0.7.

### 2. I went to the SEO > Reports page and attempted to generate a report. 

Result:

The report is never completed and logs show the `Undefined index: path` error.

![image](https://user-images.githubusercontent.com/134939/91231871-a6512a80-e6f3-11ea-90d6-76c9be16e65d.png)

I went to the CP Dashboard which did not throw a 500 error, but just showed 0%

![image](https://user-images.githubusercontent.com/134939/91232073-034ce080-e6f4-11ea-9ec7-73a43ba4c687.png)

The server shows one report directory has been created:

![image](https://user-images.githubusercontent.com/134939/91232177-37280600-e6f4-11ea-8227-4624970884ed.png)

### 3. I updated SEO Pro to v2.0.8 and cleared the stache

I checked the Reports page. The first report still shows 0% and the loader just spins forever.

Attempting to generate another report throws the following Javascript error:

![image](https://user-images.githubusercontent.com/134939/91232526-e06efc00-e6f4-11ea-9f19-9298956d51ae.png)

Checking the logs shows the same error as before is being triggered: `ERROR: Undefined index: path`

<img width="264" alt="Screenshot 2020-08-25 at 17 08 49" src="https://user-images.githubusercontent.com/134939/91232896-a9e5b100-e6f5-11ea-9d65-dae8005de030.png">

### 4. Looked like maybe a caching issue so I ran `php artisan optimise; php please stache:clear; php please static:clear`

I am now able to successfully generate a report, the previous reports still spin and show 0%

I removed all the report and tried again, I am able to successfully generate a report.

**Comment by jesseleite** [comment]
Ah an incomplete report.  I assume the dashboard widget and report listing is fixed in 2.0.10 though, even with a bad/incomplete report?

**Comment by pryley** [comment]
I'm unsure as I deleted the faulty reports after upgrading to v2.0.8.
